src/ui/map.js

```diff
@@ -1117,36 +1118,29 @@ class Map extends Camera {
     }
 
     /**
-     * Adds a listener that will be called only once to a specified event type.
-     *
-     * @method
-     * @name once
-     * @memberof Map
-     * @instance
-     * @param {string} type The event type to add a listener for.
-     * @param {Function} listener (optional) The function to be called when the event is fired once.
-     *   The listener function is called with the data object passed to `fire`,
-     *   extended with `target` and `type` properties. If the listener is not provided,
-     *   returns a Promise that will be resolved when the event is fired once.
-     * @returns {Map} `this` | Promise
-     */
-
-    /**
-     * Adds a listener that will be called only once to a specified event type occurring on features in a specified style layer.
+     * Adds a listener that will be called only once to a specified event type,
+     * optionally limited to events occurring on features in a specified style layer.
      *
      * @param {string} type The event type to listen for; one of `'mousedown'`, `'mouseup'`, `'click'`, `'dblclick'`,
      * `'mousemove'`, `'mouseenter'`, `'mouseleave'`, `'mouseover'`, `'mouseout'`, `'contextmenu'`, `'touchstart'`,
      * `'touchend'`, or `'touchcancel'`. `mouseenter` and `mouseover` events are triggered when the cursor enters
      * a visible portion of the specified layer from outside that layer or outside the map canvas. `mouseleave`
      * and `mouseout` events are triggered when the cursor leaves a visible portion of the specified layer, or leaves
      * the map canvas.
-     * @param {string} layerId The ID of a style layer. Only events whose location is within a visible
+     * @param {string} layerId (optional) The ID of a style layer. Only events whose location is within a visible
```

Comment:
Maybe it's worth expanding the descriptions of `layerId` to note that when it's not provided, the listener will be triggered by the corresponding event happening anywhere on the map.

Reply:
That's super helpful. Added in b00322750 🏆